src/wazuh_modules/wm_syscollector.c

```diff
@@ -44,29 +47,41 @@ syscollector_sync_message_func syscollector_sync_message_ptr = NULL;
 long syscollector_sync_max_eps = 10;    // Database syncrhonization number of events per seconds (default value)
 int queue_fd = 0;                       // Output queue file descriptor
 
+static bool is_shutdown_process_started() {
+    w_mutex_lock(&sys_shutdown_control_mutex);
+    bool ret_val = shutdown_process_started;
+    w_mutex_unlock(&sys_shutdown_control_mutex);
+    return ret_val;
+}
+
 static void wm_sys_send_diff_message(/*const void* data*/) {
     // const int eps = 1000000/syscollector_sync_max_eps;
     // Sending deltas is disabled due to issue: 7322 - https://github.com/wazuh/wazuh/issues/7322
     // wm_sendmsg(eps, queue_fd, data, WM_SYS_LOCATION, SYSCOLLECTOR_MQ);
  }
 
 static void wm_sys_send_dbsync_message(const void* data) {
-    if(!os_iswait()) {
+    if(!os_iswait() && !is_shutdown_process_started()) {
         const int eps = 1000000/syscollector_sync_max_eps;
         if (wm_sendmsg(eps, queue_fd, data, WM_SYS_LOCATION, DBSYNC_MQ) < 0) {
 #ifdef CLIENT
             mterror(WM_SYS_LOGTAG, "Unable to send message to '%s' (wazuh-agentd might be down). Attempting to reconnect.", DEFAULTQUEUE);
 #else
             mterror(WM_SYS_LOGTAG, "Unable to send message to '%s' (wazuh-analysisd might be down). Attempting to reconnect.", DEFAULTQUEUE);
 #endif
-            if ((queue_fd = StartMQ(DEFAULTQUEUE, WRITE, INFINITE_OPENQ_ATTEMPTS)) < 0) {
-                mterror(WM_SYS_LOGTAG, "Unable to connect to '%s'", DEFAULTQUEUE);
-            } else {
-                mtinfo(WM_SYS_LOGTAG, "Successfully reconnected to '%s'", DEFAULTQUEUE);
-                if (wm_sendmsg(eps, queue_fd, data, WM_SYS_LOCATION, DBSYNC_MQ) < 0) {
-                    mterror(WM_SYS_LOGTAG, "Unable to send message to '%s' after a successfull reconnection...", DEFAULTQUEUE);
+            // This part of the code is mutually exclusive, since this call is made from multiple threads, and this
```

Comment:
Since this method is being called by multiple threads it's necessary this particular portion of code to be mutually exclusive. When one thread is successfully reconnected, the other ones will make use of it.